ext/standard/basic_functions.c

```diff
@@ -1214,18 +1214,27 @@ PHP_FUNCTION(flush)
 /* {{{ Delay for a given number of seconds */
 PHP_FUNCTION(sleep)
 {
-	zend_long num;
+	double seconds;
+	struct timespec php_req, php_rem;
 
 	ZEND_PARSE_PARAMETERS_START(1, 1)
-		Z_PARAM_LONG(num)
+		Z_PARAM_DOUBLE(seconds)
 	ZEND_PARSE_PARAMETERS_END();
 
-	if (num < 0) {
+	if (seconds < 0) {
 		zend_argument_value_error(1, "must be greater than or equal to 0");
 		RETURN_THROWS();
 	}
 
-	RETURN_LONG(php_sleep((unsigned int)num));
+	php_req.tv_sec = (time_t) ((uint64_t)(seconds * 1000000000) / 1000000000);
+	php_req.tv_nsec = (long) ((uint64_t)(seconds * 1000000000) % 1000000000);
+	if (!nanosleep(&php_req, &php_rem)) {
```

Comment:
This needs to check for `HAVE_NANOSLEEP` now as its a requirement for `sleep()` to function, which some systems may not have

Reply:
thanks - I have added a fallback to `sleep`